### PR TITLE
docs: update new client generation instructions

### LINKED ISSUE #9902: New client library generation needs to be updated with regard to api-shortname
Currently, `new-client.py` uses `api-shortname` for directory creation. However, `api-shortname` is not always a unique value. We should determine a new methodology for creating the directory using a unique name.

For example: https://github.com/googleapis/google-cloud-java/pull/9896

### PR DESCRIPTION
**Pull request by alicejli**
Fixes #9902

### CODE REVIEW

File: generation/new_client/README.md [hunk 1]
```diff
@@ -299,6 +304,24 @@ generated Google Maps Routes API Java client library.
   --distribution-name="com.google.maps:google-maps-routing"
 ```
 
+If `maps-routing` was already in use by an existing client library as an `api_shortname`, you
+would determine a unique `destination_name` for the new library's subdirectory as well as a unique `distribution_name` for the artifact to be published (confirm with the service team).
+Let's say `maps-routing-gps` is the unique subdirectory and artifact name that has been decided
+You would then use the following command:
```
Comment (suztomo) [inline generation/new_client/README.md:310]: Which part is special?

File: generation/new_client/README.md [hunk 2]
```diff
@@ -136,9 +136,14 @@ The example in this README uses AlloyDB's [Cloud Drop](https://github.com/google
 
 ### API short name
 
-For convenience of the subsequent commands, define a variable for API short name.
+For convenience of the subsequent commands, define a variable for API short name. This will be used to generate the `distribution_name` as well as the sub-directory within this repo.
```
Comment (suztomo) [inline generation/new_client/README.md:139]: Not just these, but many params use it as their default values.

File: generation/new_client/README.md [hunk 3]
```diff
@@ -137,8 +137,15 @@ The example in this README uses AlloyDB's [Cloud Drop](https://github.com/google
 ### API short name
 
 For convenience of the subsequent commands, define a variable for API short name.
+This value will be used to generate the `distribution_name` as well as the sub-directory for the client library within this repo. It is also the default value for many additional parameters, so take care to select a unique and reasonable value.
```
Comment (suztomo) [inline generation/new_client/README.md:140]: Can you use bullet points?

File: generation/new_client/README.md [hunk 4]
```diff
@@ -137,8 +137,15 @@ The example in this README uses AlloyDB's [Cloud Drop](https://github.com/google
...
+
 The corresponding value in the Cloud Drop page is `api_short_name`.
 
+> [!IMPORTANT]
+> `api_short_name` is not always unique across client libraries.
+> In the instance that the `api_short_name` is already in use by an existing client library, determine a unique name and append it to the `--destination-name`.
+> See example under [Advanced Options](#advanced-options).
+
 Example: `alloydb`
```
Comment (suztomo) [inline generation/new_client/README.md:149]: This is not an example of "the `api_short_name` is already in use by an existing client library"

File: generation/new_client/README.md (hunk 3 above)
Comment (alicejli) [reply generation/new_client/README.md]: Good idea - updated

File: generation/new_client/README.md (hunk 4 above)
Comment (alicejli) [reply generation/new_client/README.md]: Sorry that example should have been above the `Note`. I added a comprehensive example about what to do with duplicate api_short_names below.

File: generation/new_client/README.md [hunk 5]
```diff
@@ -137,10 +137,20 @@ The example in this README uses AlloyDB's [Cloud Drop](https://github.com/google
 ### API short name
 
 For convenience of the subsequent commands, define a variable for API short name.
+This value will be used by default to generate the following:
+* `distribution_name`
```
Comment (suztomo) [inline generation/new_client/README.md:141]: Line 268 says it's "-"

```suggestion
* `distribution-name`
```

Would you check which is correct?

File: generation/new_client/README.md [hunk 6]
```diff
@@ -137,10 +137,20 @@ The example in this README uses AlloyDB's [Cloud Drop](https://github.com/google
...
+* `destination-name`
+* `proto_path`
```
Comment (suztomo) [inline generation/new_client/README.md:143]:
```suggestion
* `proto-path`
```

According to line 279. Can you check which is correct?

Comment (suztomo) [inline generation/new_client/README.md:143]: Reading the section starting from line 154, I couldn't get how the default value would be constructed for proto-path using api_shortname. Would you clarify that?

File: generation/new_client/README.md [hunk 7]
```diff
@@ -312,6 +322,34 @@ generated Google Maps Routes API Java client library.
...
 ```
+### Example with duplicate api_short_name
+
+Let's say you get a new library request where the Cloud Drop value for `api_short_name` is `maps-routing`.
+
+You discover that `maps-routing` is already in use by an existing client library!
+
+You need to determine a unique `destination_name` for the new library's subdirectory as well as a unique `distribution_name` for the artifact to be published (confirm these values with the service team on the buganizer ticket before proceeding).
+
+You will **still use** the non-unique `api_short_name` for the `api_short_name` flag.
```
Comment (suztomo) [inline generation/new_client/README.md:333]: In this example, why is it important to use maps-routing-gps in api_shortname?
(An alternative is maps-routing-gps as api_shortname.)

File: generation/new_client/README.md [hunk 8]
```diff
@@ -312,6 +322,34 @@ generated Google Maps Routes API Java client library.
...
+
+There is no hard and fast rule for determining a unique name, so some discussion will be necessary.
+
+Let's say that after some discussion, `maps-routing-gps` is selected as a suitable unique subdirectory name and artifact name.
```
Comment (suztomo) [inline generation/new_client/README.md:337]: I see the artifact name is different from "maps-routing-gps".

**Review by suztomo — APPROVED** [review]

**Pull request merged**